app/services/web-vitals.ts

```diff
@@ -10,18 +10,19 @@ export default class WebVitalsService extends Service {
   @service declare fastboot: FastBootService;
 
   static getConnectionSpeed(): string {
-    // @ts-ignore
-    return 'connection' in navigator && navigator['connection'] && 'effectiveType' in navigator['connection']
-      ? navigator['connection']['effectiveType']
-      : '';
+    if ('connection' in navigator && navigator.connection && typeof navigator.connection === 'object') {
+      if ('effectiveType' in navigator.connection && typeof navigator.connection.effectiveType === 'string') {
+        return navigator.connection.effectiveType;
```

Comment:
TypeScript is [now smarter with checking types](https://devblogs.microsoft.com/typescript/announcing-typescript-5-8/#granular-checks-for-branches-in-return-expressions) in conditional expressions, therefore this function needed to be refactored to be more TS-safe. Also removed unnecessary `@ts-ignore` statements.